src/mgmt_conn.c

```diff
@@ -1025,9 +1025,13 @@ uzfs_zvol_execute_async_command(void *arg)
 			async_task->status = ZVOL_OP_STATUS_OK;
 		}
 
-		mutex_enter(&zinfo->main_zv->rebuild_mtx);
-		zinfo->is_snap_inprogress = 0;
-		mutex_exit(&zinfo->main_zv->rebuild_mtx);
+		mutex_enter(&async_tasks_mtx);
+		if (async_task->conn_closed == B_FALSE) {
+			mutex_enter(&zinfo->main_zv->rebuild_mtx);
```

Comment:
shall we revert order of lock?
async_tasks_mtx seems to be having lock for shorter time, so better to take it as second lock.

Reply:
its better to hold rebuild_mtx lock for shorter period.